metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/v1/SystemInfoController.java

```diff
@@ -0,0 +1,161 @@
+import com.linkedin.metadata.system_info.SystemPropertiesInfo;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * REST controller for system information endpoints.
+ *
+ * <p>Provides comprehensive system information including:
+ *
+ * <ul>
+ *   <li>Spring component status (GMS, MAE Consumer, MCE Consumer)
+ *   <li>System configuration properties with metadata
+ *   <li>Property source information and filtering statistics
+ * </ul>
+ *
+ * <p>All endpoints return pretty-printed JSON for better readability in debugging/admin scenarios.
+ * This follows the same pattern used by /config endpoints in the codebase.
+ *
+ * <p><strong>Security Note:</strong> These endpoints expose system configuration data. Sensitive
+ * properties (passwords, secrets, keys) are automatically redacted as ***REDACTED***.
+ */
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/openapi/v1/system-info")
+@Tag(
+    name = "System Information",
+    description = "APIs for retrieving system configuration and component information")
+public class SystemInfoController {
+
+  private final SystemInfoService systemInfoService;
+  private final ObjectMapper objectMapper;
+
+  @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+  @Operation(
+      summary = "Get complete system information",
+      description =
+          "Retrieves comprehensive system information including Spring components and system properties with metadata. "
+              + "Returns pretty-printed JSON for better readability. Sensitive properties are automatically redacted.")
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "Successfully retrieved system information"),
+        @ApiResponse(responseCode = "500", description = "Internal server error")
+      })
+  public ResponseEntity<String> getSystemInfo() {
+    try {
+      log.debug("Request received for complete system information");
+      SystemInfoResponse response = systemInfoService.getSystemInfo();
+
+      // Return pretty-printed JSON for better readability in debugging/admin scenarios.
+      // This follows the same pattern used by /config endpoints in the codebase.
+      String json = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(response);
+      return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(json);
```

Comment:
I used pretty printed output, this way it was easier for me to look at the output as well, but doing pretty printed makes the code here a bit more verbose. Alternatively we could've enable pretty print for everything (I think there is some setting through object mapper).